### PR TITLE
Feature/upgrade to latest springboot 2

### PR DESCRIPTION
**Pull request by Telamon-m**

### CODE REVIEW

File: src/test/java/io/github/stepio/cache/caffeine/MultiConfigurationCacheManagerCustomCaffeineTest.java [hunk 1]
```diff
@@ -60,7 +59,7 @@ public void testCreateNativeCaffeineCacheWithCustomCaches() {
 
         final Object etalon = aCustom;
         await().atMost(300, TimeUnit.MILLISECONDS)
-                .until(() -> this.cachedDataHolder.newCachedCustomObject() != etalon);
```
Comment (stepio) [inline src/test/java/io/github/stepio/cache/caffeine/MultiConfigurationCacheManagerCustomCaffeineTest.java:63]: Checking by address highlights that the cache returns exactly the same object. Any particular reason for this change? Same for a few other occurrences below.

File: src/main/java/io/github/stepio/cache/caffeine/CaffeineSpecResolver.java [hunk 2]
```diff
@@ -29,7 +29,7 @@ public void setEnvironment(Environment environment) {
 
     public String getCaffeineSpec(String name) {
         String custom = this.environment.getProperty(composeKey(name));
-        if (StringUtils.isEmpty(custom)) {
```
Comment (stepio) [inline src/main/java/io/github/stepio/cache/caffeine/CaffeineSpecResolver.java:32]: Because it's deprecated, I see. Thanks!

**Review by stepio — COMMENTED** [review]
Thanks, good work! Just a minor clarification about the tests

File: src/main/java/io/github/stepio/cache/caffeine/CaffeineSpecResolver.java (hunk 2 above)
Comment (Telamon-m) [reply src/main/java/io/github/stepio/cache/caffeine/CaffeineSpecResolver.java:32]: Yes, correct - it was deprecated.

File: src/test/java/io/github/stepio/cache/caffeine/MultiConfigurationCacheManagerCustomCaffeineTest.java (hunk 1 above)
Comment (Telamon-m) [reply src/test/java/io/github/stepio/cache/caffeine/MultiConfigurationCacheManagerCustomCaffeineTest.java:63]: This is a very hard and coupled assumption. What if the cache would e. g. have a flyweight pattern integrated?
I came across this through IntelliJ's standard code inspections.
Imho equals is fine. However you are more familar with the code - so if you like I would revert the comparisons.

**Pull request merged**